Expensify/App
# Add Group DMs

Adds Group DMs

cc @shawnborton to help our with the styling. Working well on all platforms. 

I didn't bother to try figuring out how to inline the text input. It seems a little tricky to pull off since it would need to have a width that grows with the content but also wraps when it gets to be a certain length. But maybe you have some ideas on how to pull that off.

### Fixed Issues
Fixes https://github.com/Expensify/Expensify/issues/144291

### Tests
**Test that participants can be removed from the group chat list**
1. Search for a user 
1. Tap "Add" button and verify the UI changes to the group DM mode with a "Go" button
1. Remove the user by clicking on the user pill
1. Verify they are removed and the UI reverts back to the previous state

**Test that pressing "Go" can start a group chat**
1. Search for a user 
1. Tap "Add" button and verify the UI changes to the group DM mode with a "Go" button
1. Verify that user is no longer in the list
1. Add another user by tapping on "Add"
1. Tap "Go" and verify a group chat is started between the users

**Test that tapping on a user row while composing a group chat will start the chat between all users**
1. Search for a user 
1. Tap "Add" button and verify the UI changes to the group DM mode with a "Go" button
1. Verify that user is no longer in the list
1. Add another user by tapping on "Add"
1. Select a final user but do not tap "Add" and instead tap the user row itself
1. Verify a chat between the final user and all users in the group users list begins

**Test that a group DM can only have 8 participants**
1. Keep adding users until you see the warning message and cannot add another user

### Screenshots
**Web**
<img width="435" alt="Screenshot 1511" src="https://user-images.githubusercontent.com/32969087/97066417-3849a800-156a-11eb-8640-765d4e88417b.png">
<img width="391" alt="Screenshot 1512" src="https://user-images.githubusercontent.com/32969087/97066420-3aac0200-156a-11eb-9595-7ae8d064e618.png">
<img width="439" alt="Screenshot 1513" src="https://user-images.githubusercontent.com/32969087/97066422-3d0e5c00-156a-11eb-9f4b-ee929e42f141.png">

**iOS**
<img width="387" alt="Screenshot 1514" src="https://user-images.githubusercontent.com/32969087/97066681-2832c800-156c-11eb-9b66-f0e62c8b6602.png">
<img width="395" alt="Screenshot 1515" src="https://user-images.githubusercontent.com/32969087/97066683-29fc8b80-156c-11eb-8369-36c03d3d1349.png">
<img width="400" alt="Screenshot 1516" src="https://user-images.githubusercontent.com/32969087/97066686-2bc64f00-156c-11eb-9133-a1b781e1f1c6.png">

**Android**
<img width="426" alt="Screenshot 1517" src="https://user-images.githubusercontent.com/32969087/97066737-b5761c80-156c-11eb-8cdb-0c77ad2cd647.png">
<img width="424" alt="Screenshot 1518" src="https://user-images.githubusercontent.com/32969087/97066739-b73fe000-156c-11eb-8460-e37970274996.png">
<img width="411" alt="Screenshot 1519" src="https://user-images.githubusercontent.com/32969087/97066741-b9a23a00-156c-11eb-972d-faf38b7eef5f.png">
